### PR TITLE
Adding a notion of taskType in ShallowTrace. The idea is to set a typ…

### PR DESCRIPTION
**Pull request by ssodhanilinkedin**
…e (string) to tasks such as timerTask, withSideEffectTask, remoteTask etc. This tag would be displayed in trace visualization.

### CODE REVIEW

File: src/com/linkedin/parseq/Task.java [hunk 1]
```diff
@@ -229,6 +229,7 @@
     ArgumentUtil.requireNotNull(func, "function");
     final Task<Task<R>> nested = map("map: " + desc, func);
     nested.getShallowTraceBuilder().setSystemHidden(true);
+    nested.getShallowTraceBuilder().setTaskType("nestedTask");
```
Comment (jodzga) [inline src/com/linkedin/parseq/Task.java]: I think we should drop redundant "Task" suffix from the task types e.g.:
"nestedTask" -> "nested"
"withSideEffectTask" -> "withSideEffect"
"timeoutTask" -> "timeout"
"withTimeoutTask" -> "withTimeout"

File: src/com/linkedin/parseq/trace/codec/json/JsonTraceDeserializer.java [hunk 2]
```diff
@@ -90,6 +90,9 @@ public static Trace deserialize(final JsonNode rootNode) throws IOException {
       final ResultType resultType = ResultType.valueOf(getTextField(traceNode, JsonTraceCodec.TRACE_RESULT_TYPE));
       shallowBuilder.setResultType(resultType);
 
+      if (traceNode.get(JsonTraceCodec.TRACE_TASK_TYPE) != null)
```
Comment (jodzga) [inline src/com/linkedin/parseq/trace/codec/json/JsonTraceDeserializer.java]: Is there a test that covers it?

Comment (ssodhanilinkedin) [inline src/com/linkedin/parseq/trace/codec/json/JsonTraceDeserializer.java]: Added tests.

**Comment by jodzga** [comment]
Ship it!

**Pull request merged**